### PR TITLE
Codefix for SA1106

### PR DESCRIPTION
**Pull request by hvanbakel**
Codefix for SA1106, #315

### CODE REVIEW

**Comment by hvanbakel** [comment]
A review on the label fix would be nice, I haven't used those in years (fortunately).

File: StyleCop.Analyzers/StyleCop.Analyzers.Test/ReadabilityRules/SA1106UnitTests.cs [hunk 1]
```diff
@@ -132,6 +215,11 @@ public void TestMethod()
             await this.VerifyCSharpDiagnosticAsync(testCode, expected, CancellationToken.None).ConfigureAwait(false);
```
Comment (Noryoko) [inline StyleCop.Analyzers/StyleCop.Analyzers.Test/ReadabilityRules/SA1106UnitTests.cs]: Please add code fix tests.

File: StyleCop.Analyzers/StyleCop.Analyzers/ReadabilityRules/SA1106CodeFixProvider.cs [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+                    .Concat(new[] { labeledStatement.WithStatement(parentBlock.Statements.Skip(index + 1).FirstOrDefault()) })
+                    .Concat(parentBlock.Statements.Skip(index + 2));
+                var newParentBlock = parentBlock.WithStatements(SyntaxFactory.List(newStatements));
+                newRoot = root.ReplaceNode(parentBlock, newParentBlock);
+            }
+
+            return Task.FromResult(document.WithSyntaxRoot(newRoot));
+        }
+    }
+}
```
Comment (Noryoko) [inline StyleCop.Analyzers/StyleCop.Analyzers/ReadabilityRules/SA1106CodeFixProvider.cs]: New line please.

File: StyleCop.Analyzers/StyleCop.Analyzers/ReadabilityRules/SA1106CodeFixProvider.cs [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+        private static Task<Document> RemoveSemicolonFromTypeAsync(Document document, SyntaxNode root, TypeDeclarationSyntax typeDeclaration)
+        {
+            return Task.FromResult(document.WithSyntaxRoot(root.ReplaceNode(typeDeclaration, typeDeclaration.WithoutTrailingTrivia())));
+        }
+
+        private static Task<Document> RemoveEmptyStatementAsync(Document document, SyntaxNode root, EmptyStatementSyntax node)
+        {
+            var newRoot = root;
+
+            if (node.Parent.IsKind(SyntaxKind.Block))
```
Comment (Noryoko) [inline StyleCop.Analyzers/StyleCop.Analyzers/ReadabilityRules/SA1106CodeFixProvider.cs]: I think a switch statement using `node.Parent.Kind()` would be more readable here.

**Comment by sharwell** [comment]
I'd like to see the following additional test cases:

1. Two consecutive labels:

    ```csharp
    label1:
    label2:
        ;
        Console.WriteLine();
    ```

2. A `switch` statement with multiple `case` clauses and a `default` clause.

    ```csharp
    switch (default(int))
    {
    case 0:
        ;
        break;

    case 1:
    case 2:
        ;
        break;

    default:
        ;
        break;
    }
    ```

3. Each test case in `TestMemberAsync` should be updated to test the fix as well.

**Comment by sharwell** [comment]
I sent pull request hvanbakel/StyleCopAnalyzers#1 to finish off this pull request. Once @hvanbakel merges it we should be able to move forward with this. :smile:

**Pull request merged**